data-access-layer/iaas/GcpClient.js

```diff
@@ -30,6 +30,76 @@ class GcpClient extends BaseCloudClient {
     );
   }
 
+  getDiskMetadata(diskCid, zone) {
+    return Promise.try(() =>
+      this.computeClient
+      .zone(zone)
+      .disk(diskCid)
+      .get()
+      .then(diskData => diskData[0].metadata)
+      .then(metadata => {
+        return {
+          volumeId: metadata.name,
+          size: metadata.sizeGb,
+          zone: metadata.zone.substring(metadata.zone.lastIndexOf('/') + 1),
+          type: metadata.type.substring(metadata.type.lastIndexOf('/') + 1),
+          extra: {
+            tags: metadata.labels,
+            type: metadata.type
+          }
+        };
+      })
+    );
+  }
+
+  createDiskFromSnapshot(snapshotId, zone, opts = {}) {
+    return new Promise((resolve, reject) => {
+      const diskName = this.getRandomDiskId();
+      const options = {
+        sourceSnapshot: `global/snapshots/${snapshotId}`,
+        description: 'disk created via service fabrik for bosh restore',
+        type: opts.type || `projects/${this.settings.projectId || this.settings.credentials.project_id}/zones/${zone}/pd-ssd`,
+        labels: _.assign({}, opts.tags || {}, {
+          operation: 'bosh restore'
```

Comment:
May be `restore` only   or `restore by service fabrik`. Just a thought.

Reply:
will remove this altogether so that only "createdBy" is the default tag always and everything else has to be supplied by the caller